src/pages/index.js

```diff
@@ -59,26 +123,59 @@ const profileTitleInput = document.querySelector("#profile-title-input");
 const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
+const profileAvatar = document.querySelector("#profile-image-id");
 const profileEditForm = document.forms["edit-profile-modal__form"];
 const addCardFormElement = document.forms["add-card-modal__form"];
-const cardlistEl = document.querySelector("card-list-id");
+const avatarModalForm = document.forms["edit-avatar-modal__form"];
 
 const userInfo = new UserInfo({
   profileTitle,
   profileDescription,
+  profileAvatar,
 });
 
+const avatarModal = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: (inputValue) => {
+    console.log(inputValue);
+    avatarModal.setLoading(true, "Saving");
+    api
+      .avatarModal(inputValue)
+      .then((info) => {
+        userInfo.setUserInfo(info);
+      })
+      // .then((res) => {
+      //   return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+      // })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        avatarModal.setLoading(false, "Saving");
+      });
+  },
+});
+
+const avatarEditButton = document.querySelector("#edit-avatar-button");
+avatarEditButton.addEventListener("click", () => {
+  avatarModal.open();
+});
+avatarModal.setEventListeners();
+
 /**Event Handlers */
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data.Name, data.Description);
-  profileEditModal.close();
-}
-
-function handleAddCardFormSubmit(cardData) {
-  console.log(cardData);
-  renderCard(cardData, cardlistEl);
-  addCardModal.close();
-  addCardValidator.disableButton();
+  profileEditModal.setLoading(true, "Saving...");
+  api
+    .editprofileInfo({ name: data.Name, about: data.Description })
+    .then((res) => {
+      userInfo.setUserInfo(res.name, res.about);
```

Comment:
`      userInfo.setUserInfo(res.name, res.about);
`
* I can't change the profile data on the page


You should pass the whole `res` object here


```js
    .then((res) => {
      userInfo.setUserInfo(res);

```